fagiServer/src/test/java/com/fagi/handler/inputhandler/FriendRequestServerTests.java

```diff
@@ -0,0 +1,101 @@
+package com.fagi.handler.inputhandler;
+
+import com.fagi.handler.ConversationHandler;
+import com.fagi.handler.InputHandler;
+import com.fagi.model.*;
+import com.fagi.model.messages.lists.FriendList;
+import com.fagi.model.messages.message.TextMessage;
+import com.fagi.responses.AllIsWell;
+import com.fagi.responses.NoSuchUser;
+import com.fagi.responses.Response;
+import com.fagi.worker.InputAgent;
+import com.fagi.worker.OutputAgent;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+import java.util.Collections;
+
+import static org.mockito.Mockito.*;
+
+class FriendRequestServerTests {
+    private InputHandler inputHandler;
+    private OutputAgent outputAgent;
+    private InputAgent inputAgent;
+    private Data data;
+    private User user;
+
+    @BeforeEach
+    void setup() {
+        data = Mockito.mock(Data.class);
+        inputAgent = Mockito.mock(InputAgent.class);
+        outputAgent = Mockito.mock(OutputAgent.class);
+        var conversationHandler = new ConversationHandler(data);
+
+        inputHandler = new InputHandler(inputAgent, outputAgent, conversationHandler, data);
+        user = Mockito.mock(User.class);
+        var username = "bob";
+
+        doReturn(username).when(user).getUserName();
+        doReturn(user).when(data).getUser(Mockito.any());
+        doReturn(username).when(inputAgent).getUsername();
+    }
+
+    @Test
+    void whenRequestFriendFails_ShouldResultInResponseNotBeingAFriendList() {
+        var friendReq = new FriendRequest("new friend", new TextMessage("Hullo me friend", user.getUserName(), 42));
+
+        doReturn(new NoSuchUser()).when(user).requestFriend(Mockito.any(), Mockito.any());
+
+        inputHandler.handleInput(friendReq);
+
+        var argumentCaptor = ArgumentCaptor.forClass(Response.class);
+        Mockito.verify(outputAgent, times(1)).addResponse(argumentCaptor.capture());
+
+        Assertions.assertAll(
```

Comment:
In this case you are testing that you get the element NoSuchUser back in the outputAgent. Test whether you receive the same object and  not some other object that is not an AllIsWell